src/ServerRequestFactory.php

```diff
@@ -60,7 +60,7 @@ public static function fromGlobals(
         $files   = static::normalizeFiles($files ?: $_FILES);
         $headers = static::marshalHeaders($server);
 
-        if (null === $cookies && array_key_exists('cookie', $headers)) {
+        if (null === $cookies && isset($headers['cookie'])) {
```

Comment:
You need to introduce tests around `null` values to avoid regressions here

Reply:
I think it's imposible to write this test becase `$headers` comes from private function which alware return strings pair array.

Reply:
No need then: nullability was just an assumption on my end.